### PR TITLE
Base composition templates for vue

### PR DESCRIPTION
**Pull request by SahilTara**
Summary of changes:
Base composition templates for vue. These need to be PR'd before the pages to allow testability of all PR branches separately. This PR isn't testable until the pages PR.

Should be reviewed after #617 

Just hit f5 and launch webts then hit generate! Dlls are in for windows and mac!
Related issues:
#199

### CODE REVIEW

**Comment by streetcoder123** [comment]
I see that this PR also includes most of the changes that were in the `Vue base` PR. Why is that so? Also, what different am I expected to see when I run the application in this PR than what I had seen from the 'Vue base' PR?

**Comment by SahilTara** [comment]
Hey, this is because the vue base PR hasnt been merged in. Once it is I will rebase or merge and the history should update. There is actually no changes possible to be seen with this PR unfortunately. It is only possible to see through subsequent PRs but this PR allows the pagea PRs to be tested individually.

**Comment by streetcoder123** [comment]
That makes sense. Thanks

**Review by Tanya0609 — APPROVED** [review]
Compared with each file to the corresponding file in React and no much difference. So these changes are good. 

Also, Juan mentioned about two tiny formatting issues in another PR that is based on this one. Since all those PRs by Sahil are interdependent on each other and those formatting issues also exist on current codebase for React. I will suggest to fix formatting issues in a future PR all together.

File: templates/Web/_composition/Vue/Page.Vue.AddHome/src/router/index_postaction.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿export default new VueRouter({
+  routes: [
+    //^^
+    //{[{
+    { path:"/", redirect: "/Param_HomePageName" }
+    //}]}
+  ]
+});
```
Comment (jasmingoh) [inline templates/Web/_composition/Vue/Page.Vue.AddHome/src/router/index_postaction.js:8]: I'm comparing the structure to the template for React, so I'm wondering why VueRouter is required? It seems like we didn't have any ReactRouter.

Comment (SahilTara) [reply templates/Web/_composition/Vue/Page.Vue.AddHome/src/router/index_postaction.js:8]: Oh haha good question its done differently in vue, that's the main reason.

Look at this guide: https://medium.com/codingthesmartway-com-blog/vue-js-routing-with-vue-router-4c428fabb078

and these docs for more info! https://router.vuejs.org/guide/#javascript

Then check this file out:
https://github.com/microsoft/WebTemplateStudio/blob/5306b002acaa62349bf03904482730212f2e48fd/templates/Web/Projects/VueDefault/src/main.js

I completely understand the confusion here though! Hope these resources help!

**Review by jasmingoh — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Pull request merged**